src/system/RoutesModule/Entity/RouteEntity.php

```diff
@@ -59,18 +59,20 @@ public function __construct()
      */
     public function getPathWithBundlePrefix($container = null)
     {
-        if (!isset($this->options['zkNoBundlePrefix']) || !$this->options['zkNoBundlePrefix']) {
-            $bundle = $this->getBundle();
-
-            if (!\ServiceUtil::hasContainer()) {
-                \ServiceUtil::setContainer($container);
-            }
+        if (isset($this->options['zkNoBundlePrefix']) && $this->options['zkNoBundlePrefix']) {
+            // return path only
+            return $this->path;
+        }
 
-            $modinfo = \ModUtil::getInfoFromName($bundle);
+        $bundle = $this->getBundle();
 
-            return '/' . $modinfo['url'] . $this->path;
+        if (!\ServiceUtil::hasContainer()) {
+            \ServiceUtil::setContainer($container);
         }
 
-        return $this->path;
+        $modinfo = \ModUtil::getInfoFromName($bundle);
+
+        // return path prepended with bundle prefix
+        return '/' . $modinfo['url'] . $this->path;
```

Comment:
you already have the bundle in line 67 above. so do, `$bundle->getMetaData()->getUrl();`